plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -0,0 +1,304 @@
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    return await this.scheduleFn();
+  }
+
+  private createScheduleFn(schedule: TaskRunner): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return schedule.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.options.logger.child({
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(error);
+          }
+        },
+      });
+    };
+  }
+
+  async refresh(logger: Logger) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const { org, repoSearchPath } = parseUrl(this.options.target);
+
+    const { headers } = await this.githubCredentialsProvider.getCredentials({
+      url: this.options.target,
+    });
+
+    const client = graphql.defaults({
+      baseUrl: this.options.gitHubConfig.apiBaseUrl,
+      headers,
+    });
+
+    const startTimestamp = Date.now();
+
+    logger.info(`Reading GitHub repositories from ${this.options.target}`);
+
+    const expectedEntityFiles = this.options.files;
+
+    const { repositories } = await getOrganizationEntities(
+      client,
+      org,
+      expectedEntityFiles,
+    );
+
+    const matching = repositories.filter(
+      r =>
+        !r.isArchived &&
+        repoSearchPath.test(r.name) &&
+        r.defaultBranchRef?.name,
+    );
+
+    const duration = ((Date.now() - startTimestamp) / 1000).toFixed(1);
+
+    logger.info(
+      `Read ${repositories.length} GitHub repositories (${matching.length} matching the pattern) in ${duration} seconds`,
+    );
+
+    const files = expectedEntityFiles.map(fileName => ({
+      name: fileName,
+      key: camelCase(fileName),
+    }));
+
+    const repoEntities: Array<LocationSpec | Array<LocationSpec>> = [];
+
+    for (const entity of matching) {
+      for (const { name: fileName, key } of files) {
+        const entityObj: any = entity[key as keyof typeof entity];
+        if (entityObj) {
+          const locationEntity = this.createLocationSpec({
+            url: entity.url,
+            branch: entity.defaultBranchRef?.name || '',
+            file: fileName,
+          });
+          repoEntities.push(locationEntity);
+        }
+      }
+    }
+
+    const entityLocations = repoEntities.flat();
+
+    if (entityLocations.length === 0) {
+      logger.info(`No valid entities found by ${this.getProviderName()}.`);
+      return;
+    }
+
+    logger.info(
+      `Adding ${entityLocations.length} valid entities to the catalog`,
+    );
+
+    await this.connection.applyMutation({
+      type: 'full',
+      entities: entityLocations.map(location => ({
+        locationKey: this.getProviderName(),
```

Comment:
I think the locationKey that this entity provider provides should be the same that is used for the current entities that got added via the GithubDiscoveryProcessor so the migration to the entity provider should be trivial, and should work without rebuilding your catalog

Reply:
Please excuse my ignorance but I'm not sure what you are asking for here exactly. Are you suggesting that I change the providerName value to the same as the GithubDiscoveryProcessor?

Reply:
@kissmikijr the previous implementation was a processor which was based on a static location entity from the config ingested by ConfigLocationEntityProvider. I think what you suggest will not work. The processor itself was not adding a locationKey. Entities are not "owned" in the same way as at providers. Hence, I assume even if we attempt to replicate this here somehow, it will be hard for the provider to claim the entities.

I think it is better to define a proper locationKey and at the migration, remove the old data and let the new provider re-ingest it. This is how I dealt with other processor to provider migrations before, too.

@brentg-telus I think some explanation about migration steps would be great (e.g., at the changeset and then referenced at deprecation warning log for those still using the to be deprecated processor).

Reply:
@pjungermann hm, yea a hard reset will definitely work, and probably the cleanest thing to do.
However I thinkwe have an opportunity here to be able to switch over to the provider without migrations if we provide the same locationKey that is used. As I check in the database in the `refresh_state` table, you can see that every row has a locationKey. Which is constructed from the LocationSpec like: `<type>:<target>`. 

So I am thinking if we would make the locationKey to be `github-discovery:<target>`, the same the new provider could live next to the processor so there would not need migration.

@Rugvip can you take a quick glance at this? Could this work ?

Reply:
I just checked it at our DB.

For a Location entity created from static "github-discovery" Location (app-config.yaml), e.g. with a target `https://github.com/org-name/repo-name/blob/master/catalog-info.yaml`, the locationKey will be `<type>:<target>` from the static location entry like `github-discovery:https://github.com/org-name/*/blob/-/catalog-info.yaml`.

I don't think we should try to maintain the same locationKey.

Reply:
To confirm, we are leaving this as is?